src/components/Common/AutoFocusInput.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <b-form-input
+    :ref="refName"
+    v-model="value"
+    v-bind="$attrs"
+    @blur="looseFocus"
+    data-auto-focus="true"
+  />
+</template>
+
+<script>
+export default {
+  name: 'AutoFocusInput',
+  props: {
+    name: {
+      type: String,
+      required: true
+    },
+    initialValue: {
+      type: String,
+      default: ''
+    }
+  },
+  computed: {
+    refName() {
+      return `auto-focus-input-${this.name}`
+    }
+  },
+  watch: {
+    value () {
+      this.$emit('input', this.value)
+    }
+  },
+  data () {
+    return {
+      value: this.initialValue,
+      isFocus: false
+    }
+  },
+  methods: {
+    // AutoFocusInput is only activated
+    enable () {
+      const tagType = document.activeElement.tagName
+
+      // If we are currently selecting another input
+      if (tagType === 'INPUT' && !document.activeElement.dataset.autoFocus) {
+        return false
+      }
+
+      return true
+    },
+    looseFocus() {
+      this.isFocus = false
+    },
+    listenKeypress () {
```

Comment:
you can directly listen for keys events with a `v-on` directive on components
https://vuejs.org/v2/guide/events.html#Key-Modifiers

Reply:
Yes but I want to catch every event so that why the handler is on the `document` object.